android/source/VideoLocker/src/org/edx/mobile/social/SocialFactory.java

```diff
@@ -12,9 +13,19 @@
     
     public static ISocial getInstance(Activity activity, int type) {
         if (type == TYPE_GOOGLE) {
-            return new GoogleOauth2(activity);
+            if (Config.getInstance().getThirdPartyTraffic().isGoogleEnabled()) {
```

Comment:
Yes. Actually <code>LoginActivity</code> calls a few methods of <code>ISocial</code> for Google Login and for Facebook Login. Instead of having each call behind a <code>if</code> condition, I thought of having empty implementation which does nothing.